src/hooks/useDirectusAuth.tsx

```diff
@@ -6,7 +6,11 @@ import { UserType } from '@directus/sdk';
  * Possible states of the authentication.
  * @defaultValue 'loading'
  */
-export type AuthStates = 'loading' | 'authenticated' | 'unauthenticated';
+export enum AuthStates {
+  LOADING = 'loading',
```

Comment:
Are you sure that "loading" is the right state? I would expect here something like unknown. WDYT?

Reply:
The loading state is used together with the autologin prop. When that is set to true the AuthState has to be determined via checkAuth first.

Reply:
You mean that because the feature is named "autologin" then the "loading" starting state is the right choice?

Reply:
Nevermide, I see your new commit now. Sounds good!

Reply:
The state occures only at the start when the AuthState has to be loaded and validated via the directus Server. So I think loading is in order.